Framework/API/src/SpectrumInfo.cpp

```diff
@@ -132,6 +132,16 @@ Kernel::V3D SpectrumInfo::position(const size_t index) const {
   return newPos / static_cast<double>(dets.size());
 }
 
+/// Returns the phi angle (in radians) for the spectrum with given index
+double SpectrumInfo::phi(const size_t index) const {
+  if (isMonitor(index))
+    throw std::logic_error("phi is not defined for monitors.");
+  Kernel::V3D pos = position(index);
+  double p, dummy;
+  pos.getSpherical(dummy, dummy, p);
+  return p * M_PI / 180.;
```

Comment:
Would it be better to replace lines 140-142 with `return atan2(pos[1], pos[0]);`?